api/src/components/publication/controller.ts

```diff
@@ -378,6 +377,8 @@ export const getPDF = async (
     const redirectToPreview = event.queryStringParameters?.redirectToPreview === 'true';
     const publicationId = event.pathParameters.id;
     const publication = await publicationService.get(publicationId);
+    const s3Endpoint =
+        process.env.STAGE === 'local' ? process.env.LOCALSTACK_SERVER : 'https://s3.eu-west-1.amazonaws.com';
```

Comment:
This is not needed. You can import s3 from /lib/s3.ts and then just use {s3.config.endpoint}